### PR TITLE
feat(ci.jenkins, trusted.ci and cert.ci) Terminology update

### PR DESCRIPTION
**Pull request by dduportal**
Because no more Jenkins or "build" `master`: ony Jenkins controllers.

### CODE REVIEW

**Comment by dduportal** [comment]
Please note that the puppet agent had been disabled in ci.j, trusted.j and cert.ci.j: once this is merged, a full "puppet agent --test --noop" will be done on each to ensure there is no breaking.

**Comment by dduportal** [comment]
Putting back as draft to allow delivering #2260 and #2259 + fixing https://github.com/jenkins-infra/helpdesk/issues/3039

**Comment by dduportal** [comment]
- Disabled (again) the puppet agent on the 3 controllers in production, after deploying #2259 #2260 and #2261
- Rebased the PR
- Local tests are OK. CI builds are OK.
- Local test with `vagrant up jenkins::controller` gives me the same container (with a Jenkins controller) as the one I had with `vagrant up jenkins::master` previously


=> Ready for review, but wait for merge please.

File: dist/profile/manifests/jenkinscontroller.pp [hunk 1]
```diff
@@ -1,19 +1,19 @@
 #
-# Profile for configuring the bare necessities to running a Jenkins master
+# Profile for configuring the bare necessities to running a Jenkins controller
```
Comment (lemeurherve) [inline dist/profile/manifests/jenkinscontroller.pp]:
```suggestion
# Profile for configuring the bare necessities to run a Jenkins controller
```

File: dist/profile/templates/azure.ci.jenkins.io/agents.yaml.erb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 jenkins:
-  ## No builds on master
+  ## No builds on controller
```
Comment (lemeurherve) [inline dist/profile/templates/azure.ci.jenkins.io/agents.yaml.erb]:
```suggestion
  # No builds on controller
```
(nit)

File: dist/role/manifests/jenkins/controller.pp [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#
+# Role for a Jenkins controller
```
Comment (lemeurherve) [inline dist/role/manifests/jenkins/controller.pp]:
```suggestion
# Role for a Jenkins controller
```

File: manifests/site.pp [hunk 4]
```diff
@@ -70,28 +70,23 @@
 
 node 'azure.ci.jenkins.io' {
   sshkeyman::hostkey { ['azure.ci.jenkins.io']: }
-  include role::jenkins::master
-}
```
Comment (lemeurherve) [inline manifests/site.pp]: Why do these lines have been removed? Typo?

File: manifests/site.pp [hunk 5]
```diff
@@ -70,28 +70,23 @@
...
-
-node /^agent-\d+$/ {
-  include role::jenkins::agent
```
Comment (lemeurherve) [inline manifests/site.pp]: Or is it because we don't want "agent-*" node anymore?

**Review by lemeurherve — COMMENTED** [review]
Nits + a question about node "agent-*"

File: manifests/site.pp (hunk 4 above)
Comment (dduportal) [reply manifests/site.pp]: not a typo, but a cleanup that Ithought it was on another commit: lemme fix (I prefer separating PRs here)

Comment (dduportal) [reply manifests/site.pp]: fixed

**Review by lemeurherve — DISMISSED** [review]

**Pull request merged**